tests/base/weight_sharing_module.py

```diff
@@ -0,0 +1,18 @@
+from torch import nn
+
+from tests.base import SimpleModule
+
+
+class WeightSharingModule(SimpleModule):
```

Comment:
all the tests are being shifted to use `BoringModel`. Also maybe keep this definition in the same file it is being used.
```suggestion
class WeightSharingModule(BoringModel):
```

Reply:
Used `SimpleModule` because `BoringModel` does not call `forward` in `training_step` and `validation_step`. I would have to redefine those. Maybe we can fix that in `BoringModel`?

Also the `WeightSharingModule` is used twice, and defining it twice didn't make sense.

Reply:
ok cool... seems like SimpleModule is using the similar data/api as of BoringModel. then it's fine. For WeightSharingModel you can define it at the top of the file where it's being used.